rpmio/rpmpgp_internal.c

```diff
@@ -1150,9 +1153,13 @@ rpmRC pgpVerifySignature(pgpDigParams key, pgpDigParams sig, DIGEST_CTX hashctx)
      * done all we can, return NOKEY to indicate "looks okay but dunno."
      */
     if (key && key->alg) {
+	if (key->tag != PGPTAG_PUBLIC_KEY &&
+	    key->tag != PGPTAG_PUBLIC_SUBKEY)
+	    goto exit; /* not a public key */
 	pgpDigAlg sa = sig->alg;
 	pgpDigAlg ka = key->alg;
-	if (sa && sa->verify) {
+	if (sa && sa->verify &&
+	    sig->pubkey_algo == key->pubkey_algo) {
```

Comment:
Another broken indentation here. A (continued) conditional on the same indentation level as the next actual line of code is a cardinal sin with no excuses or exceptions because it's *so* easy to misread.

 `indent -kr` would show/fix these.